lib/sinicum/imaging/max_size_converter.rb

```diff
@@ -12,22 +12,29 @@ def initialize(configuration)
         @format = configuration['format'] || 'jpeg'
       end
 
-      def convert(infile_path, outfile_path, extension)
-        x = device_pixel_size(@x)
-        y = device_pixel_size(@y)
-        
+      def convert(infile_path, outfile_path, extension, srcset_option = nil)
```

Comment:
You should just pass in the highres factor instead of `srcset_option`. When it's `nil`, use the normal value.